### PR TITLE
Newsletters: add 89 (2020-03-18)

### PR DESCRIPTION
**Pull request by harding**
- [x] Services/Clients section @bitschmidty 
- [x] `LND #3821` @dongcarl 

Another short one this week.

### CODE REVIEW

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #16902][] changes consensus code to fix an inefficiency
+  in the parsing of `OP_IF` and related opcodes.  In legacy and segwit
+  v0 script, the inefficiency isn't believed to cause any significant
+  problems, but the proposal for [tapscript][topic tapscript] would
```
Comment (adamjonas) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: This sentence is a couple of ideas. Suggest breaking it up.

```suggestion
  problems. Still, the proposal for [tapscript][topic tapscript] would
```

**Comment by dongcarl** [comment]
An attempt:

LND #3821 adds anchor commitments for LN channels and enables it by default if both participating nodes of a channel signal support. Anchor commitment transactions can be fee bumped unilaterally by either party, which is useful because commitment transactions might be broadcast a long time after they commit to their on-chain feerate. See the "Anchor outputs" topic for more details.

**Comment by harding** [comment]
@dongcarl LGTM, thanks!  Added to the newsletter as written except that I removed the last sentence and placed its link in the first sentence (without otherwise editing that sentence).

Also ACK @bitschmidty's section.  Thanks!

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  make it possible for an attacker to use the inefficiency to create blocks with transactions
+  that could take a large amount of CPU to verify.  Fixing the
+  inefficiency now reduces the number of changes that need to be made in the
+  proposed schnorr, taproot, and tapscript soft fork.  For more
+  information, see the Bitcoin Core PR Review Club [meeting notes][club
+  #16902] about this PR and a [related PR][Bitcoin Core #18002] also
+  merged this week.
+
+- [LND #3821][] adds [anchor commitments][topic anchor outputs] for
+  LN channels and enables it by default if both participating nodes of a
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: I'm not sure 'it' is correct here. It refers to "anchor commitments" which is plural, so suggests to me that we should use 'them'. Alternatively, "LND 3821 implements anchor commitments for enables this feature for new LN channels by default ..." or similar

**Comment by jnewbery** [comment]
Pushed a commit to link to the github version of the scaling book

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+- **Coinbase withdrawal transactions now using batching:** Coinbase has rolled
+out [batch withdrawals][coinbase batching blog] that they estimate will reduce
+their load on the Bitcoin network by 50%. Instead of each withdrawal payment
+generating a single onchain transaction, multiple payments will be
+[combined into a single transaction][scaling payment batching] once every 10 minutes.
+
+- **Bitstamp supports bech32:** Bitstamp users can now benefit from using native
```
Comment (jonatack) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: Perhaps hoist "deposits and withdrawals" to the bitstamp title for clear quick comparison to the derebit title while skim reading.

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+{% include linkers/issues.md issues="16902,3821,3963,18002,1319" %}
+[lnd op safety]: https://github.com/lightningnetwork/lnd/blob/master/docs/safety.md
+[segura email]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-March/002586.html
+[watchtower bolt]: https://github.com/sr-gi/bolt13/blob/master/13-watchtowers.md
+[club #16902]: https://bitcoincore.reviews/16902/
+[news75 watchtower]: /en/newsletters/2019/12/04/#proposed-watchtower-bolt
+[news85 ln stuck]: /en/newsletters/2020/02/19/#c-lightning-3500
+[coinbase batching blog]: https://blog.coinbase.com/coinbase-rolls-out-bitcoin-transaction-batching-5f6d09b8b045
+[bitstamp bech32 blog]: https://www.bitstamp.net/article/weve-added-support-bech32-bitcoin-addresses-bitsta/
+[deribit bech32 withdrawal tweet]: https://twitter.com/DeribitExchange/status/1234904442169851909
```
Comment (jonatack) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: Tried to find a better link for this on deribit's blog but it seems to be down currently (which surely explains why the tweet is being used here instead).

File: _posts/en/newsletters/2020-03-18-newsletter.md (hunk 2 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-03-18-newsletter.md]: Agree, was about to suggest s/it/this feature/

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  that could take a large amount of CPU to verify.  Fixing the
+  inefficiency now reduces the number of changes that need to be made in the
+  proposed schnorr, taproot, and tapscript soft fork.  For more
+  information, see the Bitcoin Core PR Review Club [meeting notes][club
+  #16902] about this PR and a [related PR][Bitcoin Core #18002] also
+  merged this week.
+
+- [LND #3821][] adds [anchor commitments][topic anchor outputs] for
+  LN channels and enables it by default if both participating nodes of a
+  channel signal support. Anchor commitment transactions can be fee
```
Comment (jonatack) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: Looked at the changeset to verify that anchor commitments are auto-enabled if both participants signal support"; comment and code seems to be here, lines 1114-1126: https://github.com/lightningnetwork/lnd/pull/3821/files#diff-691ca64693e704fd2e5080998f3e640eR1114

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  bumped unilaterally by either party, which is useful because
+  commitment transactions might be broadcast a long time after they
+  commit to their on-chain feerate.
+
+- [LND #3963][] adds detailed [documentation][lnd op safety] about how
+  to use LND safely.
```
Comment (jonatack) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: This looks like a PR of truly high value to users that might be worth a bit more pizzazz here. FWIW I like the name "lnd Operational Safety Guidelines".

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+- [Eclair #1319][] implements the same solution as described in
```
Comment (jonatack) [inline _posts/en/newsletters/2020-03-18-newsletter.md]: ISTM the PR title or a short description could be useful here, e.g. Eclair #1319 "Funder reserve for future fee increase" or descriptive equivalent, before writing that it is the same solution as described in Newsletter #85.

File: _posts/en/newsletters/2020-03-18-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+- [Bitcoin Core #16902][] changes consensus code to fix an inefficiency
+  in the parsing of `OP_IF` and related opcodes.  In legacy and segwit
+  v0 script, the inefficiency isn't believed to cause any significant
+  problems. However, the proposal for [tapscript][topic tapscript] would
+  make it possible for an attacker to use the inefficiency to create blocks with transactions
+  that could take a large amount of CPU to verify.  Fixing the
+  inefficiency now reduces the number of changes that need to be made in the
+  proposed schnorr, taproot, and tapscript soft fork.  For more
+  information, see the Bitcoin Core PR Review Club [meeting notes][club
+  #16902] about this PR and a [related PR][Bitcoin Core #18002] also
```
[the remark this answers is not in the document]
Comment (jonatack) [reply _posts/en/newsletters/2020-03-18-newsletter.md]: Agree.

**Comment by jonatack** [comment]
(verified the links so far as well)

**Comment by jnewbery** [comment]
@jonatack 
> Question: what plans for the review club feature (and frequency)?

Plan is one per month, on the second week, with a summary of the most interesting discussion over the previous month's meetings.

That gives the following schedule:
- 1st week: no special section
- 2nd week: Bitcoin Core PR Review Club
- 3rd week: Changes to services and client software
- 4th week: Selected Q&A from Bitcoin StackExchange

Let me know if you want to write some of the Bitcoin Core PR Review Club sections.

File: _posts/en/newsletters/2020-03-18-newsletter.md (hunk 4 above)
Comment (bitschmidty) [reply _posts/en/newsletters/2020-03-18-newsletter.md]: There wasn’t anything on the blog when I looked (and it was up)

**Comment by jonatack** [comment]
> Plan is one per month, on the second week, with a summary of the most interesting discussion over the previous month's meetings.
> 
> Let me know if you want to write some of the Bitcoin Core PR Review Club sections.

Sure :+1:

**Pull request merged**